test/build.py

```diff
@@ -137,27 +137,31 @@ def ignore(_src, names):
     for js_file in glob.glob(os.path.join(HTML_TESTS_DIR, '*.js')):
         shutil.copy(js_file, out_dir)
 
-def build_html_from_js(js_html_dir, html_dir):
+def build_html_from_js(js_html_dir, html_dir, use_sync):
     for js_file in glob.glob(os.path.join(js_html_dir, '*.js')):
         js_filename = os.path.basename(js_file)
         html_filename = js_filename + '.html'
         html_file = os.path.join(html_dir, html_filename)
+        js_harness = "sync_index.js" if use_sync else "async_index.js"
         with open(html_file, 'w+') as f:
             content = HTML_HEADER.replace('{PREFIX}', './js/harness') \
-                                 .replace('{WPT_PREFIX}', WPT_URL_PREFIX)
+                                 .replace('{WPT_PREFIX}', './js/harness') \
+                                 .replace('{JS_HARNESS}', js_harness)
             content += "        <script src=./js/{SCRIPT}></script>".replace('{SCRIPT}', js_filename)
             content += HTML_BOTTOM
             f.write(content)
 
-def build_html(html_dir, js_dir):
+def build_html(html_dir, js_dir, use_sync):
     print("Building HTML tests...")
 
     js_html_dir = os.path.join(html_dir, 'js')
 
-    build_html_js(js_html_dir, js_dir)
+    build_html_js(js_html_dir, js_dir, include_harness=True)
+    for js_file in glob.glob(os.path.join(js_html_dir, '*.js')):
+        wrap_single_test(js_file)
```

Comment:
ditto, this now happens after every call to `build_html_js`, so it could be part of this function.